### PR TITLE
rename from beman_iterator to beman_iterator_interace

### PR DESCRIPTION
**Pull request by changkhothuychung**
#7059

Previous PR: https://github.com/compiler-explorer/compiler-explorer/pull/7060

Per discussion in https://github.com/beman-project/iterator_interface/issues/6, we would like to rename this library to beman_iterator_interface

### CODE REVIEW

**Review by junlarsen — COMMENTED** [review]
If we change the library id, then any links with the old library id will break. DO you think it's fine to just keep `beman_iterator` and simply change its .name?

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -3858,6 +3858,12 @@ libs.belleviews.description=A library of C++ views that just works for all basic
 libs.belleviews.versions.trunk.version=trunk
 libs.belleviews.versions.trunk.path=/opt/compiler-explorer/libs/belleviews/trunk/sources
 
+libs.beman_iterator_interface.name=beman_iterator_interface
```
Comment (neatudarius) [inline etc/config/c++.amazon.properties:3861]: same here, if we can do the updates to be `beman.itereator_interface`.

CC: @changkhothuychung @junlarsen - more context here https://github.com/compiler-explorer/infra/pull/1453/files#r1836720720

Comment (neatudarius) [inline etc/config/c++.amazon.properties:3861]:
```suggestion
libs.beman_iterator_interface.name=beman.iterator_interface
```

same here, at least `.name` should be `beman.itereator_interface`.

Check conversation here: 
https://github.com/compiler-explorer/infra/pull/1453#discussion_r1836773092. Maybe the key `libs.beman_iterator_interface.name` can be converted to `libs.beman.iterator_interface.name`, but let's wait the conclusion there.

File: etc/config/c++.amazon.properties [hunk 2]
```diff
@@ -3858,6 +3858,12 @@ libs.belleviews.description=A library of C++ views that just works for all basic
...
+libs.beman_iterator_interface.versions=trunk
+libs.beman_iterator_interface.url=https://github.com/beman-project/iterator_interface
+libs.beman_iterator_interface.versions.trunk.version=trunk
+libs.beman_iterator_interface.versions.trunk.path=/opt/compiler-explorer/libs/beman_iterator_interface/main/include
+
 libs.beman_optional26.name=beman_optional26
```
Comment (neatudarius) [inline etc/config/c++.amazon.properties:3867]:
```suggestion
libs.beman_optional26.name=beman.optional26
```

Please apply same updates for beman.optional26 (https://github.com/beman-project/Optional26?tab=readme-ov-file#bemanoptional26-c26-extensions-for-stdoptional)

**Review by neatudarius — APPROVED** [review]
LGTM 👍

**Comment by changkhothuychung** [comment]
> LGTM 👍

There is a failed CI test, do you know what that error is?

**Comment by neatudarius** [comment]
@changkhothuychung , I think we need to test it with a local deployment of Compiler Explorer.

I think I found a bug in - https://github.com/compiler-explorer/infra/pull/1453.

Current PR should be OK.

**Comment by neatudarius** [comment]
> > LGTM 👍
> 
> There is a failed CI test, do you know what that error is?

I do not know compiler infra details, but please follow suggestions from the other PR (https://github.com/compiler-explorer/infra/pull/1453/files#r1837618987) and we should be able to test both diffs/PRs with same deployment.

**Pull request merged**